src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -885,6 +885,11 @@ fn new_unit_dep_with_profile(
         .resolve()
         .is_public_dep(parent.pkg.package_id(), pkg.package_id());
     let features_for = unit_for.map_to_features_for(artifact);
+    let artifact_target_for_features = if let FeaturesFor::ArtifactDep(target) = features_for {
```

Comment:
When used as part of a `let`, I'd suggest `match` as opposed to an `if-let-else`.